### PR TITLE
Move physical-item material data to single property

### PR DESCRIPTION
**Pull request by stwlam**

### CODE REVIEW

File: src/module/item/physical/data.ts [hunk 1]
```diff
@@ -42,11 +42,9 @@ interface PhysicalSystemSource extends ItemSystemSource {
         value: string;
     };
     containerId: string | null;
-    preciousMaterial: {
-        value: Exclude<PreciousMaterialType, "dragonhide" | "grisantian-pelt"> | null;
-    };
-    preciousMaterialGrade: {
-        value: PreciousMaterialGrade | null;
+    material: {
```
Comment (CarlosFdez) [inline src/module/item/physical/data.ts:45]: Why is this not ItemMaterialData? The type seems to match completely.

File: src/module/item/physical/sheet.ts [hunk 2]
```diff
@@ -105,7 +105,7 @@ class PhysicalItemSheetPF2e<TItem extends PhysicalItemPF2e> extends ItemSheetPF2
             return result;
         }, {} as MaterialSheetData["materials"]);
 
-        const value = material.precious ? `${material.precious.type}|${material.precious.grade}` : "";
+        const value = material.type && material.grade ? `${material.type}|${material.grade}` : "";
```
Comment (CarlosFdez) [inline src/module/item/physical/sheet.ts:108]: I see that adamantine earlier in the PR defaults grade to standard if its unset. Should this do the same?

**Comment by CarlosFdez** [comment]
Neato!

File: src/module/item/physical/data.ts (hunk 1 above)
Comment (stwlam) [reply src/module/item/physical/data.ts]: I think I had some idea that one or both properties would be optional in source but decided to keep them both present but possibly `null`.

File: src/module/item/physical/sheet.ts (hunk 2 above)
Comment (stwlam) [reply src/module/item/physical/sheet.ts:108]: That was to handle adamantine coming from a special ability, like Adamantine Strikes, which doesn't specify a grade.

**Review by CarlosFdez — APPROVED** [review]

**Pull request merged**